nifi-toolkit/nifi-toolkit-cli/src/main/java/org/apache/nifi/toolkit/cli/impl/result/registry/VersionedFlowSnapshotsResult.java

```diff
@@ -0,0 +1,77 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.toolkit.cli.impl.result.registry;
+
+import org.apache.commons.lang3.Validate;
+import org.apache.nifi.registry.flow.VersionedFlowSnapshot;
+import org.apache.nifi.toolkit.cli.api.WritableResult;
+import org.apache.nifi.toolkit.cli.impl.util.JacksonUtils;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.PrintStream;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Iterator;
+
+/**
+ * Result for a list of VersionedFlowSnapshots.
+ *
+ * If this result was created with a non-null exportDirectoryName, then the write method will ignore
+ * the passed in PrintStream, and will write the serialized snapshot to the given directory.
+ * The file name will be generated from the flow name and its version.
+ *
+ * If this result was created with a null exportDirectoryName, then the write method will write the
+ * serialized snapshots to the given PrintStream.
+ */
+public class VersionedFlowSnapshotsResult implements WritableResult<Iterator<VersionedFlowSnapshot>> {
+    private static final String FILE_NAME_PREFIX = "toolkit_registry_export_all";
+    private static final String EXPORT_FILE_NAME = "%s/%s_%s_%s_%d";
+    private static final String SEPARATOR = "_";
+    private static final String REPLACEMENT = "-";
+    private final Iterator<VersionedFlowSnapshot> versionedFlowSnapshots;
+    private final String exportDirectoryName;
+
+    public VersionedFlowSnapshotsResult(final Iterator<VersionedFlowSnapshot> versionedFlowSnapshots, final String exportDirectoryName) {
+        this.versionedFlowSnapshots = versionedFlowSnapshots;
+        this.exportDirectoryName = exportDirectoryName;
+        Validate.notNull(this.versionedFlowSnapshots);
+    }
+
+    @Override
+    public Iterator<VersionedFlowSnapshot> getResult() {
+        return versionedFlowSnapshots;
+    }
+
+    @Override
+    public void write(final PrintStream output) throws IOException {
+        while (versionedFlowSnapshots.hasNext()) {
+            final VersionedFlowSnapshot versionedFlowSnapshot = versionedFlowSnapshots.next();
+            if (exportDirectoryName != null) {
+                final String bucketName = versionedFlowSnapshot.getBucket().getName().replaceAll(SEPARATOR, REPLACEMENT);
+                final String flowName = versionedFlowSnapshot.getFlow().getName().replaceAll(SEPARATOR, REPLACEMENT);
+                final int version = versionedFlowSnapshot.getSnapshotMetadata().getVersion();
+                final String exportFileName = String.format(EXPORT_FILE_NAME, exportDirectoryName, FILE_NAME_PREFIX, bucketName, flowName, version);
+                try (final OutputStream resultOut = Files.newOutputStream(Paths.get(exportFileName))) {
+                    JacksonUtils.write(versionedFlowSnapshot, resultOut);
+                }
```

Comment:
If the output directory doesn't exist, currently the export fails, but it isn't very clear why...
```
#> registry export-all-flows --outputDirectory /tmp/registry-export -verbose

All buckets collected...


All flows collected...


All flow versions collected...


ERROR: /tmp/registry-export/toolkit_registry_export_all_B1_Resuable_1
```

With adding `-verbose` we can see:
```
java.nio.file.NoSuchFileException: /tmp/registry-export/toolkit_registry_export_all_B1_Resuable_1
	at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:92)
	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111)
	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:116)
	at java.base/sun.nio.fs.UnixFileSystemProvider.newByteChannel(UnixFileSystemProvider.java:219)
	at java.base/java.nio.file.spi.FileSystemProvider.newOutputStream(FileSystemProvider.java:478)
	at java.base/java.nio.file.Files.newOutputStream(Files.java:220)
	at org.apache.nifi.toolkit.cli.impl.result.registry.VersionedFlowSnapshotsResult.write(VersionedFlowSnapshotsResult.java:69)
	at org.apache.nifi.toolkit.cli.impl.command.CommandProcessor.processCommand(CommandProcessor.java:256)
	at org.apache.nifi.toolkit.cli.impl.command.CommandProcessor.processGroupCommand(CommandProcessor.java:233)
	at org.apache.nifi.toolkit.cli.impl.command.CommandProcessor.process(CommandProcessor.java:188)
	at org.apache.nifi.toolkit.cli.CLIMain.runInteractiveCLI(CLIMain.java:124)
	at org.apache.nifi.toolkit.cli.CLIMain.main(CLIMain.java:68)
```

Match we should add something here like:
```
catch(Exception e) {
  throw new RuntimeException("Unable to write flow snapshot to: " + exportFileName, e);
}
```
?